mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1150,17 +1172,11 @@ def _generate_model_predictions(self):
                 self.y_pred = np.concatenate(y_pred_list, axis=0)
             elif isinstance(sample_pred, list):
                 self.y_pred = sum(y_pred_list, [])
-            # handle if sample_pred is a pandas series
```

Comment:
Is there a reason we just merge all the values? 
Would it even makes sense to compute the metrics based on that? 

https://github.com/mlflow/mlflow/pull/9661/files#diff-1b3ab8e1152a42ab0bd3d4e01c7621b37a95aa02fff19642a5abf23f75b6e688R423
I have changed a bit of the code above. Let me know your thoughts

Reply:
we need to merge so that we can call `predict` on individual rows in a way that is transparent to the user.